packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -1017,15 +1017,15 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
         minimumSize: MaterialStateProperty.resolveWith<Size?>((Set<MaterialState> states) {
           final double? effectiveMaximumWidth =
               effectiveMenuStyle!.maximumSize?.resolve(states)?.width;
-          return Size(math.min(widget.width!, effectiveMaximumWidth ?? 0.0), 0.0);
+          return Size(math.min(widget.width!, effectiveMaximumWidth ?? widget.width!), 0.0);
         }),
       );
     } else if (anchorWidth != null) {
       effectiveMenuStyle = effectiveMenuStyle.copyWith(
         minimumSize: MaterialStateProperty.resolveWith<Size?>((Set<MaterialState> states) {
           final double? effectiveMaximumWidth =
               effectiveMenuStyle!.maximumSize?.resolve(states)?.width;
-          return Size(math.min(anchorWidth, effectiveMaximumWidth ?? 0.0), 0.0);
+          return Size(math.min(anchorWidth, effectiveMaximumWidth ?? anchorWidth!), 0.0);
```

Comment:
Removing `!` after `anchorWidth` should fix one of the linux analyzer failures.
```suggestion
          return Size(math.min(anchorWidth, effectiveMaximumWidth ?? anchorWidth), 0.0);
```